text/0000-deprecation-ember-map.md

```diff
@@ -0,0 +1,37 @@
+- Start Date: 2017-07-20
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the deprecation of the following classes:
+
+- `Ember.OrderedSet`
+- `Ember.Map`
+- `Ember.MapWithDefault`
+
+These classes need to be moved to an external addon given they are private classes and unused in Ember.js itself.
+
+# Motivation
+
+These classes have not been used in Ember itself for a while now. They have always been private but they are used in a few addons, and in particular Ember Data is using them.
+
+# Transition Path
+
+The classes will be extracted to an addon and the ones in Ember will be deprecated.
```

Comment:
@buschtoens I don't think it is. Switching to native `Map` seems like a big undertaking, especially if we are to support IE9 and family :P